src/sentry/releasehealth/metrics.py

```diff
@@ -0,0 +1,152 @@
+from datetime import datetime
+from typing import Dict, Optional, Sequence, Set
+
+from snuba_sdk import Column, Condition, Entity, Op, Query
+from snuba_sdk.expressions import Granularity
+
+from sentry.models.project import Project
+from sentry.releasehealth.base import ReleaseHealthBackend
+from sentry.sentry_metrics import indexer
+from sentry.sentry_metrics.indexer.base import UseCase
+from sentry.snuba.dataset import Dataset
+from sentry.utils.snuba import raw_snql_query
+
+
+def metric_id(org_id: int, name: str) -> int:
+    index = indexer.resolve(org_id, UseCase.TAG_KEY, name)  # type: ignore
+    assert index is not None  # TODO: assert too strong?
+    return index  # type: ignore
+
+
+def tag_key(org_id: int, name: str) -> str:
+    index = indexer.resolve(org_id, UseCase.TAG_KEY, name)  # type: ignore
+    assert index is not None
+    return f"tags[{index}]"
+
+
+def tag_value(org_id: int, name: str) -> int:
+    index = indexer.resolve(org_id, UseCase.TAG_VALUE, name)  # type: ignore
+    assert index is not None
+    return index  # type: ignore
+
+
+def reverse_tag_value(org_id: int, index: int) -> str:
+    str_value = indexer.reverse_resolve(org_id, UseCase.TAG_VALUE, index)  # type: ignore
+    assert str_value is not None
+    return str_value  # type: ignore
+
+
+class MetricsReleaseHealthBackend(ReleaseHealthBackend):
+    """Gets release health results from the metrics dataset"""
+
+    def get_current_and_previous_crash_free_rates(
+        self,
+        project_ids: Sequence[int],
+        current_start: datetime,
+        current_end: datetime,
+        previous_start: datetime,
+        previous_end: datetime,
+        rollup: int,
+        org_id: Optional[int] = None,
+    ) -> ReleaseHealthBackend.CurrentAndPreviousCrashFreeRates:
```

Comment:
Why are doing this kind of type inheritance ? I would just move the types outside of the class.. Surely they will be used in other areas of the code that use this function

Reply:
This has been fixed in master.